lib/serializer/BinaryDeserializer.cpp

```diff
@@ -62,7 +61,7 @@ void CLoadFile::openNextFile(const boost::filesystem::path & fname, int minimalV
 		{
 			logGlobal->warn("Warning format version mismatch: found %d when current is %d! (file %s)\n", serializer.fileVersion, SERIALIZATION_VERSION , fName);
 
-			auto versionptr = (char*)&serializer.fileVersion;
+			auto *versionptr = (char*)&serializer.fileVersion;
```

Comment:
I thought that clang-tidy would replace c-style casts with static_cast/reinterpret_cast?

Reply:
Not all, but I will try with version 15 too.

Reply:
If I am reading this correctly, looks to be present, including auto-fix. And available in older versions.
https://releases.llvm.org/12.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/cppcoreguidelines-pro-type-cstyle-cast.html

Reply:
Yes, it is present. But this rule is present too: ```This rule bans (T)expression only when used to perform an unsafe cast.``` Maybe clang-tidy thinks than cast is safe here.

Reply:
It might be safe here, but here it is definitely NOT safe:
https://github.com/vcmi/vcmi/pull/1561#discussion_r1100218176

Well, I guess we should look into that warning eventually...

Reply:
reverted